rclcpp/src/rclcpp/node_interfaces/node_topics.cpp

```diff
@@ -64,11 +64,19 @@ NodeTopics::create_publisher(
 
 void
 NodeTopics::add_publisher(
-  rclcpp::PublisherBase::SharedPtr publisher)
+  rclcpp::PublisherBase::SharedPtr publisher,
+  rclcpp::callback_group::CallbackGroup::SharedPtr callback_group)
```

Comment:
Hmmm, I wonder if it would be better to specify the callback_group for event callbacks as part of the optional class where the callback functions are set instead. What do you think of that idea?

Reply:
I think @ross-desmond said `CallbackGroup` is used often, which is why I didn't fold it into the options class. I'm open to changing it.

Reply:
Yeah, I'm not as concerned about exposing it to the user. I was more thinking that the callback group for publishers really only applies to the event callbacks and that for subscribers you may want to separate your event handlers and your message handler into different callback groups. Maybe I'm over thinking it though so I'm fine leaving this as is to not add more work for now if we don't think it's absolutely the right thing to do.